template.rb

```diff
@@ -1,9 +1,69 @@
 require 'shellwords'
 
+# Set Thor::Actions source path for looking up the files
+def source_paths
+  @source_paths
+end
+
+# Prepend the required paths to the source paths to make the template files in those paths available
+def use_source_path(source_path)
+  @source_paths.unshift(source_path)
+end
+
+def remote_repository
+  require 'tmpdir'
+  tempdir = Dir.mktmpdir('rails-templates')
+  at_exit { FileUtils.remove_entry(tempdir) }
+
+  git clone: [
+    '--quiet',
+    'https://github.com/nimblehq/rails-templates.git',
+    tempdir
+  ].map(&:shellescape).join(' ')
+
+  if (branch = __FILE__[%r{rails-templates/(.+)/template.rb}, 1])
+    Dir.chdir(tempdir) { git checkout: branch }
+  end
+
+  tempdir
+end
+
+def delete_test_folder
+  FileUtils.rm_rf('test')
+end
```

Comment:
This could be externalized to a separate file -> `test/template.rb`. It would just remove it the test directory 🙃 Might sound silly but it would allow externalizing this concern just like for other directories, then it could allow later on to have a flag to use Minitest if needed.